internal/manager/controllerdef.go

```diff
@@ -196,8 +196,14 @@ func setupControllers(mgr manager.Manager, proxy proxy.Proxy, c *Config) ([]Cont
 				IngressClassName: c.IngressClassName,
 			},
 		},
+		// ---------------------------------------------------------------------------
+		// Other Controllers
+		// ---------------------------------------------------------------------------
 		{
-			Enabled: c.KnativeIngressEnabled,
+			// knative is a special case because it existed before we added feature gates functionality
+			// for this controller (only) the existing --enable-controller-knativeingress flag overrides
+			// any feature gate configuration. See FEATURE_GATES.md for more information.
+			Enabled: featureGates["Knative"] || c.KnativeIngressEnabled,
```

Comment:
I think that feature gate names deserve a dedicated enum type, and they definitely deserve `const`s rather than literal values